### ISSUE TITLE
[CWM] Add loading animations

### ISSUE THREAD

**Issue by rogup** [DigitalCommons/mykomap-monolith#94]
## Description

When a search is being performed or a popup's contents are being retrieved, there aren't any animations, so it looks like the app isn't do anything, sometimes leading to double-clicks and clunky UX.

Especially a problem if the internet connection is poor or server has more load.

I think adding animations to indicated things are loading would make things feel a lot smoother, e.g. the popup could open and there could be some nice animations like this https://mui.com/material-ui/react-skeleton/#wave-example

## Acceptance Criteria

- [ ]

**Comment by ColmDC** [comment]
Can you check that this link is the correct one? https://mui.com/material-ui/react-skeleton/#wave-example

**Comment by rogup** [comment]
Yes, that's the right one. I think either pulsate or wave animations look pretty good and are commonly used

**Comment by ColmDC** [comment]
And the wave passes across the who frame the map occupies, or just the map. behind the panels?

**Comment by rogup** [comment]
@ColmDC The flows I'm imagining are:

1. User performs a search
-> an empty results panel opens immediately with the wave animation (in the panel) to indicate results are loading
-> once returned, the results are are populated in the panel and the map is updated at the same time.

2. User clicks a popup
-> an empty popup opens immediately with the wave animation (in the popup)
-> once returned, the item info is populated in the popup

**Comment by ColmDC** [comment]
Would it be needed in case 2, given how little data is loaded at that point? There is a little delay at the moment (on my current connection/version) but I was assuming that was not from pulling the data down. But if so, then it makes sense. Is a seperate case required for the initial map load?

**Comment by ColmDC** [comment]
Any thoughts @ms0ur1s ?

**Comment by ColmDC** [comment]
> It might also be nice to use the skeleton as placeholder for the missing / loading data.

Is this a third scenario you're pointing to here @ms0ur1s? or are you concurring with @rogup that the wave animation would be good to have in the pop-up before the data for is has been downloaded?

**Comment by rogup** [comment]
I'm noticing a split second delay between the triangle appearing and then the data populating, maybe this will be slightly longer for people on poor connections. But yes, maybe the better solution for popups would be to just delay the triangle appearing, so that it appears when the rest of the popup does.

I think the reason to show a skeleton would be so that there's an immediate response to the user's click, to make the app feel snappier. But I suppose this isn't actually needed, since the map will pan, so the user knows their click has done something. Also, showing a skeleton might look a bit clunky since the popup resizes depending on the amount of content, so it might flicker from small to big.
